### DISCUSSION TITLE
BYE is not sent for calls over websocket

### DISCUSSION THREAD

**Discussion by pitti98** [baresip/re#804] [General]
This may be related to #143.
We are trying to implement a websocket server, so browsers connect to libre via websocket.

Responses to an INVITE get sent just fine, but a BYE after 200 OK does not.

When I step through the code, what happens is that in sip_request_send(), because the "somethingsomething.invalid" host that I see used with websockets is neither an IP address nor a real host that can be looked up,  the first *if* that performs the request immediately is not entered, and dnsc_query() ends up being called but returns an error. So no BYE gets sent at all.

I haven't tried yet, but I suspect making calls to a websocket client will fail as well because the INVITE won't find its way to the already connected client.

At the moment I'm reading the code and the RFCs trying to undertand how it all should work, but this is quite complex and I haven't been able to come up with a quick fix for this. It seems websocket support is missing in this corner of the library, especially the websocket *server side*?
Anyway, what I think needs to happen in general is that the code must notice that it's a websocket address, and then consult the table of connected websocket clients to find the client and its connected socket.

**Comment by alfredh** [comment]
please note that SIP over Websock should be considered "Work in progress".

It is not production quality.


Can you please provide some logs for your setup ?
